federationapi/routing/join.go

```diff
@@ -278,11 +278,9 @@ func SendJoin(
 	return util.JSONResponse{
 		Code: http.StatusOK,
 		JSON: gomatrixserverlib.RespSendJoin{
-			RespState: gomatrixserverlib.RespState{
-				StateEvents: gomatrixserverlib.UnwrapEventHeaders(stateAndAuthChainResponse.StateEvents),
-				AuthEvents:  gomatrixserverlib.UnwrapEventHeaders(stateAndAuthChainResponse.AuthChainEvents),
-			},
-			Origin: cfg.Matrix.ServerName,
+			StateEvents: gomatrixserverlib.UnwrapEventHeaders(stateAndAuthChainResponse.StateEvents),
```

Comment:
We might have spotted this earlier if we had the spec readily to hand. Please add a link to the spec for this response format before the `return`.